apps/web/src/hooks/useBestAMMTrade.ts

```diff
@@ -49,14 +56,15 @@ interface Options {
   stableSwap?: boolean
   enabled?: boolean
   autoRevalidate?: boolean
+  queryArgs: GetQuoteArgs
 }
 
 interface useBestAMMTradeOptions extends Options {
   type?: 'offchain' | 'quoter' | 'auto' | 'api'
 }
 
 export function useBestAMMTrade({ type = 'quoter', ...params }: useBestAMMTradeOptions) {
```

Comment:
Let's not change the interface of our useBestAMMTrade hook.

We can put all the conversion logic inside the universal-router-sdk and use our own trade type on the UI side.

For example after we get the best trade from smart-router, we can pass that trade object to universal-router-sdk to assemble the calldata you need to interact with the smart contract